BxlAsLib.txt

```diff
@@ -0,0 +1,377 @@
+BuildXL as a library - aka. BXLLib.
+
+Problem statement
+-----------------
+- Performance
+  - Graph construction takes a long time due to DScript parsing, type checking, and evaluation.
+  - DScript frontend is not scalable to handle Office multi-architecture (mularchy) builds.

```

Comment:
>not scalable [](start = 24, length = 12)

Does not have `binary module caching` yet

Reply:
Instead of this "not scalable" bullet point, I'd say something like "graph construction is not modular, i.e., there is no way to compose multiple separate (possibly interconnected) graph fragments into a single final graph

Reply:
We have a design to do that wich is the feature I mentioned. Just not implemented yet.